src/doodle/konvaFuncs.js

```diff
@@ -0,0 +1,68 @@
+import Konva from 'konva';
+import { createElement } from 'preact';
+import { useState, useRef, useEffect} from 'preact/hooks';
+
+//TODO not sure how to enforce this type...
+/**
+ * @typedef Line
+ * @property {string} tool - The tool used to draw this line.
+ * @property {any} points
+ */
+
+
+const Canvas = ({width, height, handleMouseDown, handleMouseUp, handleMouseMove, handleMouseLeave, lines}) => {
```

Comment:
Currently, every re-render of this causes us to redraw every line. This seems kind of inefficient, so it may be worth monitoring that to see if it's causing too much lag. If so, then we'd want to think of ways to speed that up. No changes in this PR, just we should keep this in mind.